syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -347,6 +377,127 @@ static GOptionEntry no_options[] =
   { NULL, 0, 0, G_OPTION_ARG_NONE, NULL, NULL }
 };
 
+static void
+set_console_echo(gboolean new_state)
+{
+  struct termios t;
+
+  if (tcgetattr(STDIN_FILENO, &t))
+    {
+      fprintf(stderr, "syslog-ng-ctl: error while tcgetattr: %s\n", strerror(errno));
+      return;
+    }
+
+  if (new_state)
+    t.c_lflag |= ECHO;
+  else
+    t.c_lflag &= ~((tcflag_t) ECHO);
+
+  if (tcsetattr(STDIN_FILENO, TCSANOW, &t))
+    fprintf(stderr, "syslog-ng-ctl: error while tcsetattr: %s\n", strerror(errno));
+}
+
+static void
+get_password_from_stdin(gchar *buffer, gulong *length)
+{
+  printf("enter password:");
+  set_console_echo(FALSE);
+  if (-1 == getline(&buffer, length, stdin))
+    {
+      set_console_echo(TRUE);
+      fprintf(stderr, "error while reading password from terminal: %s", strerror(errno));
+      g_assert_not_reached();
+    }
+  set_console_echo(TRUE);
+  printf("\n");
+}
+
+static gboolean
+is_syslog_ng_running()
+{
+  return control_client_connect(control_client);
+}
+
+static gchar *
+fetch_next_remaining(gchar **remaining, gint *available_index)
```

Comment:
I would rename this function to indicate the incrementation of `available_index`, for example:
`consume_next_arg` or `consume_next_from_remaining`